app/views/reservations/new.html.haml

```diff
@@ -25,7 +25,14 @@
 
 = simple_form_for([@order, @order_detail, @reservation]) do |f|
   = f.error_messages
-  = render partial: 'account_field', locals: { f: f } unless @order_detail.bundled?
+
+  - unless @order_detail.bundled?
+    - accounts = @order.user.accounts_for_product(@instrument)
```

Comment:
I could definitely see it getting extracted into something; not quite sure what. One option would be a helper. Another would be a SimpleForm input. In `account_field`, you could probably just say `f.input :account, as: :account_select` and the logic could be inside an `AccountSelectInput` class. That should give you access to the object that is stored in `@order`.